### ISSUE TITLE
`linkage_print_diagram` never returns for small screen_widths

### ISSUE THREAD

**Issue by ged** [opencog/link-grammar#626]
When given a screen width argument less than some minimum (related, I suspect, to the particulars of the diagram) `linkage_print_diagram` never returns.

Here's a minimum reproduction that shows the problem for the sentence `"The flag was wet."` and a screen width of `11`:
```c
#include <locale.h>
#include <stdio.h>
#include "link-grammar/link-includes.h"

int main()
{
    Dictionary    dict;
    Parse_Options opts;
    Sentence      sent;
    Linkage       linkage;
    char *        string;
    char *        input_string = "The flag was wet.";

    setlocale(LC_ALL, "");
    opts = parse_options_create();
    dict = dictionary_create_lang("en");
    if (!dict) {
        printf ("Fatal error: Unable to open the dictionary\n");
        return 1;
    }

    sent = sentence_create(input_string, dict);
    if (sentence_split(sent, opts)) {
        printf ("Warning: Unable to tokenize the sentence\n");
        return 1;
    }
    if (sentence_parse(sent, opts)) {
        linkage = linkage_create(0, sent, opts);

        string = linkage_print_diagram(linkage, true, 12);
        printf("12 max width: %s", string);
        linkage_free_diagram(string);

        string = linkage_print_diagram(linkage, true, 11);
        printf("11 max width: %s", string);
        linkage_free_diagram(string);

        linkage_delete(linkage);
    }
    sentence_delete(sent);

    dictionary_delete(dict);
    parse_options_delete(opts);

    return 0;
}
```

Some info about my environment:
```
$ link-parser --version
Version: link-grammar-5.4.2

$ uname -vr
16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64

$ cc --version
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Thanks for link-grammar!

**Comment by ampli** [comment]
Hello @ged,
Thank you for reporting this bug.
I could repeat it in 5.4.2 with:
```
echo 'The flag was wet.'  |  link-parser -wall -width=11 | cat
```
This particular problem is already fixed in the current master repository (RC 5.4.3).

However, I noted that there is still a looping problem with width of 1 and 2, which I would like to solve.
There is no point to fix that for these widths, so I will just notify an error and return NULL.

Notes about returning NULL in that case:
Previous versions had another error condition in linkage_print_diagram(), when the diagram was "too big" for the (then) preallocated space. In that case the returned string was an error message.
But it is hard to handle programmatically errors that are so returned.
As null is already returned by the LG API in other  error cases, and even by linkage_print_diagram() when the linkage argument is NULL, I will return NULL also in that case.

**Comment by ged** [comment]
Sounds great. Thanks for the explanation!

**Comment by ampli** [comment]
Fixed in 5.4.3.
